react/week4/githubuser/src/UsersContext.js

```diff
@@ -0,0 +1,47 @@
+import React, { createContext, useState } from "react";
+
+export const UsersContext = createContext();
+
+const ContextProvider = ({ children }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [users, setUsers] = useState([]);
+  const [inputvalue, setInputValue] = useState("");
+  const [message, setMessage] = useState("");
+  const [repos, setRepos] = useState([]);
+
+  const contextValue = {
+    isLoading,
+    setIsLoading,
+    users,
+    setUsers,
+    inputvalue,
+    setInputValue,
+    message,
+    setMessage,
+    repos,
+    setRepos,
+    getGitHubUserWithFetch: () => {
+      if (inputvalue !== "") {
+        setIsLoading(true);
+        fetch(`https://api.github.com/search/users?q=${inputvalue}`)
+          .then((res) => res.json())
+          .then((data) => {
```

Comment:
When you work with promises, be sure to add a `.catch` as well for unexpected errors.

Reply:
actually i dont know should i use .catch here but after your comment then i get to know ,thank you for that.